src/main/kotlin/org/rust/ide/console/RsConsoleCodeFragmentContext.kt

```diff
@@ -10,42 +10,98 @@ import com.intellij.openapi.project.DumbService
 import com.intellij.openapi.project.Project
 import org.rust.cargo.project.model.cargoProjects
 import org.rust.lang.core.psi.*
-import org.rust.lang.core.psi.ext.descendantOfTypeStrict
+import org.rust.lang.core.psi.ext.RsNamedElement
+import org.rust.lang.core.psi.ext.block
+import org.rust.lang.core.psi.ext.childOfType
+import org.rust.lang.core.resolve.TYPES
+import org.rust.lang.core.resolve.findPrelude
+import org.rust.lang.core.resolve.processNestedScopesUpwards
 import org.rust.openapiext.toPsiFile
 
-class RsConsoleCodeFragmentContext {
+class RsConsoleCodeFragmentContext(codeFragment: RsReplCodeFragment?) {
 
-    private val topLevelElements: MutableMap<String, String> = mutableMapOf()
-    private val allCommandsText: StringBuilder = StringBuilder()
+    private val itemsNames: MutableSet<String> = mutableSetOf()
```

Comment:
Maybe use `ContainerUtil.newConcurrentSet()` and get rid of synchronized `addItemsNames`?